pkg/controller/certificates/rootcacertpublisher/publisher.go

```diff
@@ -171,8 +171,15 @@ func (c *Publisher) processNextWorkItem() bool {
 }
 
 func (c *Publisher) syncNamespace(ns string) error {
+	var err error
 	startTime := time.Now()
 	defer func() {
+		code := "500"
```

Comment:
this looks like it records a 500 code even in success and no-op cases

Reply:
Unit tests could be quite helpful here.

Reply:
done